backend/admin/service.go

```diff
@@ -2,22 +2,52 @@ package admin
 
 import (
 	"context"
+	"database/sql"
 	"encoding/json"
 	"fmt"
+	"net/url"
 
 	"connectrpc.com/connect"
+	"github.com/alecthomas/kong"
+	_ "github.com/jackc/pgx/v5/stdlib"
 
 	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
 	"github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/ftlv1connect"
 	"github.com/TBD54566975/ftl/go-runtime/encoding"
 	"github.com/TBD54566975/ftl/internal/configuration"
 	"github.com/TBD54566975/ftl/internal/configuration/manager"
 	"github.com/TBD54566975/ftl/internal/configuration/providers"
+	"github.com/TBD54566975/ftl/internal/dsn"
 	"github.com/TBD54566975/ftl/internal/log"
+	internalobservability "github.com/TBD54566975/ftl/internal/observability"
+	"github.com/TBD54566975/ftl/internal/rpc"
 	"github.com/TBD54566975/ftl/internal/schema"
 	"github.com/TBD54566975/ftl/internal/schema/schemaeventsource"
 )
 
+type Config struct {
+	Bind                 *url.URL `help:"Socket to bind to." default:"http://127.0.0.1:8896" env:"FTL_BIND"`
+	DSN                  string   `help:"DAL DSN." default:"${dsn}" env:"FTL_DSN"`
+	MaxOpenDBConnections int      `help:"Maximum number of database connections." default:"20" env:"FTL_MAX_OPEN_DB_CONNECTIONS"`
+	MaxIdleDBConnections int      `help:"Maximum number of idle database connections." default:"20" env:"FTL_MAX_IDLE_DB_CONNECTIONS"`
+}
+
+func (c *Config) SetDefaults() {
+	if err := kong.ApplyDefaults(c, kong.Vars{"dsn": dsn.PostgresDSN("ftl")}); err != nil {
+		panic(err)
+	}
+}
```

Comment:
Do we want/need `kong` stuff in here? I wonder if we could do this `kong` work in the `cli/cmd` code. Then, we might be able to remove this from the `service` layer.

Reply:
i don't feel super strongly - i think we went with calling it in `start` to centralize the logic across invocations from CLI, wdyt? since the config fields are defined with `default` field tags it doesn't feel completely out of place to me in the service layer